cypress-io/cypress
# feat(unify): spec-runner pane resizing

- Closes https://cypress-io.atlassian.net/browse/UNIFY-592 and https://cypress-io.atlassian.net/browse/UNIFY-628

### Additional details

In the unified spec runner we had not yet implemented the feature to resize the panels. This PR implements a custom solution for this. I did evaluate some existing libraries, most were overpowered and/or worked with different modal of how panels would resize, were missing some events or props that we needed. So the least complexity was to make something that's tailored to the needs we have for the spec runner page.

This PR:

- Updates VueUse (doesn't break anything so might as well keep the change in)
- Avoids repeatedly setting preferences
- Adds a new ResizablePanels component + tests (could use feedback about the readability of both to somebody with fresh eyes)
- Uses ResizablePanels to lay out the SpecRunner pages
- Adds reporter and specs list widths to saved preferences
- Fixes some small but important UI issues with spacing, overflow, and scroll behavior


### How has the user experience changed?

https://user-images.githubusercontent.com/8340719/146035862-282e7c67-59dd-4660-ada2-16069b80416d.mov

### How to test

Follow visit any spec in open mode, in any browser and resize the specs list and reporter. Verify that the following things are true:

- Specs List and Reporter both have a minimum width where they remain functional (header looks fine, controls are reachable)
- The AUT has a minimum width that is small, but does not allow the AUT to be pushed off screen by resizing the other panels (resizing the window can still cause problems, this PR doesn't address that, bu). 
- Only the AUT and the panel being resized are flexible when resizing - eg the Reporter size shouldn't change the Specs list size ever.
- On refreshing the page, or closing and restarting Cypress, the starting position of the Specs List and the Reporter should be the same.

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
